### ISSUE TITLE
Incorrect syntax highlighting for ': in type with TypeOperators extension

### ISSUE THREAD

**Issue by ryantm** [haskell/haskell-mode#1508]
Hello.

Thanks for your work on haskell-mode; it is great! This is a syntax highlighting bug report.

I am using haskell-mode version 20170406.25 from MELPA. My use-package configuration for haskell-mode is:

```elisp
(use-package haskell-mode
  :mode "\\.l?hs\\'"
  :ensure t
  :bind ("C-c ," . haskell-mode-format-imports)
  :config
  (add-hook 'haskell-mode-hook 'turn-on-haskell-doc-mode)
  (add-hook 'haskell-mode-hook 'turn-on-haskell-indentation)

  (use-package flycheck-haskell
    :defer t
    :ensure t)
  (use-package hi2
    :ensure t))
```
My entire Emacs configuration is at https://github.com/ryantm/dotfiles/tree/master/emacs/.emacs.d .

When I write 

```haskell
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE TypeOperators #-}

type B = Int
type C = '[]
type A = B ': C
```
in a file by itself, it gets syntax highlighted as:
![cropped](https://cloud.githubusercontent.com/assets/4804/26116296/84f642e4-3a17-11e7-8b86-83384d76785c.png)

which seems to show the apostrophe as un-matched and it highlights the rest of the line like it were part of a character.

Ryan

**Comment by gracjan** [comment]
Place to fix:

https://github.com/haskell/haskell-mode/blob/master/haskell-lexeme.el#L219

Any takers?

**Comment by Ptival** [comment]
I was looking to report this error but could not find it. So commenting to add "single quote" in case anyone looks it up with this name.

I have the exact same issue trying to user freer-effects, this breaks both the highlighting, and the auto-indentation for all subsequent lines.

**Comment by purcell** [comment]
Fixed by #1578.
